ctru-rs/src/smdh.rs

```diff
@@ -0,0 +1,99 @@
+use crate::services::cfgu::Language;
+use bitflags::bitflags;
+
+#[derive(Copy, Clone, Debug)]
+#[repr(C)]
```

Comment:
I wonder if using `repr(C)` will cause extra padding that throws off the byte offsets. You probably need to pass a byte buffer to libctru and then parse the bytes into a struct.